FlowCal/gate.py

```diff
@@ -334,17 +364,35 @@ def density2d(data, channels=[0,1],
     if data_ch.shape[0] <= 1:
         raise ValueError('data should have more than one event')
 
-    # Extract default bins if necessary
-    if (bins is None and hasattr(data_ch, 'hist_bin_edges')
-            and data_ch.hist_bin_edges(0) is not None
-            and data_ch.hist_bin_edges(1) is not None):
-        bins = np.array(data_ch.hist_bin_edges())
+    # If ``data_ch.hist_bins()`` exists, obtain bin edges from it if
+    # necessary.
+    if hasattr(data_ch, 'hist_bins'):
```

Comment:
`hist_bins` is a function. was thinking it would be better to check if it's callable. that might get kinda kluge, though, if you first need to check if the data object has `hist_bins` and then to check if `hist_bins` is callable. Originally thought there was a 1-test way to check if something is callable, but couldn't find it quickly.